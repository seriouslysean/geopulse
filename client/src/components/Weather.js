import React from "react";
import Loading from "../components/Loading";

class Weather extends React.Component {
  state = {
    showFarenheit: true
  };
  handleTemperatureToggle = e => {
    e.preventDefault();
    this.setState(prevState => ({
      showFarenheit: !prevState.showFarenheit
    }));
  };
  renderWeather = () => {
    const {
      display_location: { city, state, country },
      weather,
      temp_f,
      temp_c,
      icon,
      forecast_url
    } = this.props.weather;
    return (
      <div className="weather__data">
        <div
          className={`weather__temperature ${
            this.state.showFarenheit ? "weather__temperature--f" : "weather__temperature--c"
          }`}
          onClick={this.handleTemperatureToggle}
        >
          <h2>
            <span className="temp temp--f">{temp_f}&deg;F</span>
            <span className="temp temp--c">{temp_c}&deg;C</span>
          </h2>
        </div>
        <div className="weather__details">
          <h3>{weather}</h3>
          <p>
            {city}, {state} / {country}
          </p>
        </div>
      </div>
    );
  };
  renderContent = () => {
    console.log(this.props.weather);
    return (
      <div className="content">
        {!this.props.weather.hasOwnProperty("error") ? (
          this.renderWeather()
        ) : (
          <p className="text--center">No weather data found</p>
        )}
      </div>
    );
  };
  render() {
    return (
      <section className="section section--spacing section--bg-earth-light weather">
        <div className="content-container">
          <div className="weather__content">
            <h1 className="section__title">Weather</h1>

            {this.props.ready && this.props.weather ? this.renderContent() : <Loading />}

            <a rel="external" href="https://www.wunderground.com/" className="section__attribution" target="_blank">
              Powered by Weather Underground
            </a>
          </div>
        </div>
      </section>
    );
  }
}

export default Weather;
