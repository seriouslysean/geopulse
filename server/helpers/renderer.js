import React from 'react';
import serialize from 'serialize-javascript';
import { renderToString } from 'react-dom/server';
import { StaticRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { renderRoutes } from 'react-router-config';
import { Helmet } from 'react-helmet';
import AppRoutes from '../../client/src/routes/AppRoutes';

export default (req, store, context) => {
  const html = (
    <Provider store={store}>
      <StaticRouter location={req.path} context={context}>
        <div className="app-container">{renderRoutes(AppRoutes)}</div>
      </StaticRouter>
    </Provider>
  );
  const content = renderToString(html);
  const helmet = Helmet.renderStatic();

  return `
    <!DOCTYPE html>
    <html>
      <head>
        <meta charset="utf-8">
        <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no" />

        ${helmet.title.toString()}
        ${helmet.meta.toString()}

        <link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon.png">
        <link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png">
        <link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png">
        <link rel="manifest" href="/manifest.json">
        <link rel="mask-icon" href="/safari-pinned-tab.svg" color="#43578c">

        <!-- iOS -->
        <meta name="apple-mobile-web-app-capable" content="yes">
        <meta name="apple-mobile-web-app-status-bar-style" content="black">
        <meta name="apple-mobile-web-app-title" content="Geopulse">
        <link rel="apple-touch-startup-image" href="/assets/geopulse-splash.png">
      
        <!-- Other -->
        <meta name="application-name" content="Geopulse">
        <meta name="theme-color" content="#43578c">
        <meta name="format-detection" content="telephone=no">
      
        <!-- Social -->
        <meta itemprop="name" content="Geopulse">
        <meta itemprop="description" content="Get a pulse on the happenings nearby">
        <meta itemprop="image" content="https://geopulse-app.herokuapp.com/assets/geopulse-banner.png">
      
        <meta property="og:image" content="https://geopulse-app.herokuapp.com/assets/geopulse-banner.png">
        <meta property="og:image:width" content="1200">
        <meta property="og:image:height" content="628">
        <meta property="og:title" content="Geopulse">
        <meta property="og:description" content="Get a pulse on the happenings nearby">
        <meta property="og:url" content="https://geopulse-app.herokuapp.com">
      
        <meta name="twitter:card" content="summary_large_image">
        <meta name="twitter:title" content="Geopulse">
        <meta name="twitter:description" content="Get a pulse on the happenings nearby">
        <meta name="twitter:creator" content="@seriouslysean">
        <meta name="twitter:image" content="https://geopulse-app.herokuapp.com/assets/geopulse-banner.png">
        
        <link rel="stylesheet" type="text/css" href="bundle.css" />
      </head>
      <body>
        <div id="root">${content}</div>
        <script>window.INITIAL_STATE = ${serialize(store.getState())}</script>
        <script src="bundle.js"></script>
      </body>
    </html>
  `;

  res.send(html);
};
