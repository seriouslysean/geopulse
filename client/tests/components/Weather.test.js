import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import Weather from "../../src/components/Weather";
import weatherProp from "../../../data/fixtures/weather";

test("should render Weather correctly", () => {
  const wrapper = shallow(<Weather />);
  expect(wrapper).toMatchSnapshot();
});

test("should render Weather with content correctly", () => {
  const wrapper = shallow(<Weather weather={weatherProp} />);
  expect(wrapper).toMatchSnapshot();
});
