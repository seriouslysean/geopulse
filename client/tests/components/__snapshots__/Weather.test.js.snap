// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Weather correctly 1`] = `
<section
  className="section section--spacing section--bg-earth-light weather"
>
  <div
    className="content-container"
  >
    <div
      className="weather__content"
    >
      <h1
        className="section__title"
      >
        Weather
      </h1>
      <Loading />
      <a
        className="section__attribution"
        href="https://www.wunderground.com/"
        rel="external"
        target="_blank"
      >
        Powered by Weather Underground
      </a>
    </div>
  </div>
</section>
`;

exports[`should render Weather with content correctly 1`] = `
<section
  className="section section--spacing section--bg-earth-light weather"
>
  <div
    className="content-container"
  >
    <div
      className="weather__content"
    >
      <h1
        className="section__title"
      >
        Weather
      </h1>
      <Loading />
      <a
        className="section__attribution"
        href="https://www.wunderground.com/"
        rel="external"
        target="_blank"
      >
        Powered by Weather Underground
      </a>
    </div>
  </div>
</section>
`;
